### ISSUE TITLE
AnyType throws "Parameter count mismatch" exception when a dictionary value is a list

### ISSUE THREAD

**Issue by MariaCobretti** [ChilliCream/hotchocolate#2416]
I have a dictionary<string, object> property that is declared as an AnyType
reading that item works gread as long as the dictionary value does not contain a list

working result looks like:

```
"nodes": [
   "itemId: "2d669ba7e63243469f45224e1f2aa2dd"
   "properties": {
      "_ea01a2136b5848f9a866cd18fc9a3d91": "test1"
    }
]
```

not working but expected result should look like:

```
"nodes": [
   "itemId: "2d669ba7e63243469f45224e1f2aa2dd"
   "properties": {
      "_ea01a2136b5848f9a866cd18fc9a3d91": [
         "test1",
         "test2"
      ]
    }
]
```
```

"stackTrace": "   
at System.Reflection.RuntimeMethodInfo.InvokeArgumentsCheck(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)\r\n   
at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)\r\n   
at System.Reflection.RuntimePropertyInfo.GetValue(Object obj, Object[] index)\r\n   
at HotChocolate.Utilities.ObjectToDictionaryConverter.VisitObject(Object obj, Action`1 setValue, ISet`1 processed) 
in D:\\Projects\\ChilliCream\\hotchocolate\\src\\HotChocolate\\Core\\src\\Types\\Utilities\\ObjectToDictionaryConverter.cs:line 105\r\n   
at HotChocolate.Utilities.ObjectToDictionaryConverter.VisitValue(Object obj, Action`1 setValue, ISet`1 processed) 
in D:\\Projects\\ChilliCream\\hotchocolate\\src\\HotChocolate\\Core\\src\\Types\\Utilities\\ObjectToDictionaryConverter.cs:line 78\r\n   
at HotChocolate.Utilities.ObjectToDictionaryConverter.VisitList(ICollection list, Action`1 setValue, ISet`1 processed) 
in D:\\Projects\\ChilliCream\\hotchocolate\\src\\HotChocolate\\Core\\src\\Types\\Utilities\\ObjectToDictionaryConverter.cs:line 125\r\n   
at HotChocolate.Utilities.ObjectToDictionaryConverter.VisitValue(Object obj, Action`1 setValue, ISet`1 processed) 
in D:\\Projects\\ChilliCream\\hotchocolate\\src\\HotChocolate\\Core\\src\\Types\\Utilities\\ObjectToDictionaryConverter.cs:line 74\r\n   
at HotChocolate.Utilities.ObjectToDictionaryConverter.VisitObject(Object obj, Action`1 setValue, ISet`1 processed) 
in D:\\Projects\\ChilliCream\\hotchocolate\\src\\HotChocolate\\Core\\src\\Types\\Utilities\\ObjectToDictionaryConverter.cs:line 97\r\n   
at HotChocolate.Utilities.ObjectToDictionaryConverter.VisitValue(Object obj, Action`1 setValue, ISet`1 processed) 
in D:\\Projects\\ChilliCream\\hotchocolate\\src\\HotChocolate\\Core\\src\\Types\\Utilities\\ObjectToDictionaryConverter.cs:line 78\r\n   
at HotChocolate.Utilities.ObjectToDictionaryConverter.Convert(Object obj) 
in D:\\Projects\\ChilliCream\\hotchocolate\\src\\HotChocolate\\Core\\src\\Types\\Utilities\\ObjectToDictionaryConverter.cs:line 30\r\n   
at HotChocolate.Types.AnyType.TrySerialize(Object runtimeValue, Object& resultValue) 
in D:\\Projects\\ChilliCream\\hotchocolate\\src\\HotChocolate\\Core\\src\\Types\\Types\\Scalars\\AnyType.cs:line 217\r\n   
at HotChocolate.Types.ScalarType.Serialize(Object runtimeValue) 
in D:\\Projects\\ChilliCream\\hotchocolate\\src\\HotChocolate\\Core\\src\\Types\\Types\\Scalars\\ScalarType.cs:line 148\r\n   
at HotChocolate.Execution.Processing.ValueCompletion.TryCompleteLeafValue(IOperationContext operationContext, IMiddlewareContext middlewareContext, Path path, ILeafType fieldType, Object result, Object& completedResult) 
in D:\\Projects\\ChilliCream\\hotchocolate\\src\\HotChocolate\\Core\\src\\Execution\\Processing\\ValueCompletion.cs:line 409",
```

**Comment by PascalSenn** [comment]
@MariaCobretti  this should be fixed in #2997

**Comment by michaelstaib** [comment]
@onionhammer can you create a test for us in a PR showing the issue?

This is an example on how to create a repro:
https://github.com/ChilliCream/graphql-platform/blob/main/src/HotChocolate/Core/test/Types.Tests/Regressions/Issue_4811.cs
